### PR TITLE
Fix servicing branch release-6.0.x

### PR DESCRIPTION
**Pull request by heng-liu**
## Bug

Fixes: 
https://github.com/NuGet/Client.Engineering/issues/1312
https://github.com/NuGet/Client.Engineering/issues/1427
https://github.com/NuGet/Client.Engineering/issues/1303
https://github.com/NuGet/Client.Engineering/issues/971
https://github.com/NuGet/Client.Engineering/issues/1536
https://github.com/NuGet/Client.Engineering/issues/1448

Fix one of the branch in: https://github.com/NuGet/Client.Engineering/issues/1513

Regression? Last working version:

## Description

Fix release-6.0.x branch，
1. Remove dotnet format check
2. Skip Apex tests by cherry-pick https://github.com/NuGet/NuGet.Client/pull/4312
3. Cherry pick the fixes of the above issues.
 
## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [x] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Review by erdembayar — DISMISSED** [review]

File: test/EndToEnd/tests/GetPackageTest.ps1 [hunk 1]
```diff
@@ -181,6 +181,8 @@ function Test-GetPackageForProjectReturnsCorrectPackages2 {
 }
 
 function Test-GetPackageForFSharpProjectReturnsCorrectPackages {
+    [SkipTest('https://github.com/NuGet/Home/issues/12835')]
```
Comment (erdembayar) [inline test/EndToEnd/tests/GetPackageTest.ps1]: Is it correct issue? I get 404 when I open it.

Comment (heng-liu) [reply test/EndToEnd/tests/GetPackageTest.ps1]: Good catch~ I fixed it. Thanks!

**Review by erdembayar — DISMISSED** [review]

**Comment by heng-liu** [comment]
Hi @erdembayar , I just cherry-picked one more fix: https://github.com/NuGet/NuGet.Client/pull/4541.
Pls review it again. Thanks!

**Review by erdembayar — DISMISSED** [review]

**Comment by heng-liu** [comment]
Hi @erdembayar , the build finally passed. Pls review again, thanks!
Feel free to merge it if I don't do that in time. Thanks!

**Review by erdembayar — APPROVED** [review]

**Pull request merged**